pkg/resources/view_acceptance_test.go

```diff
@@ -50,158 +69,318 @@ func TestAcc_View(t *testing.T) {
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "aggregation_policy.0.entity_key.0", "ROLE_NAME")),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "row_access_policy.#", "1")),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "row_access_policy.0.policy_name", rowAccessPolicy.ID().FullyQualifiedName())),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "row_access_policy.0.on.#", "1")),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "row_access_policy.0.on.0", "ROLE_NAME")),
 				),
 			},
-			// update parameters
+			// change policies
 			{
-				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_View_basic"),
-				ConfigVariables: m2,
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("snowflake_view.test", "name", accName),
-					resource.TestCheckResourceAttr("snowflake_view.test", "statement", query),
-					resource.TestCheckResourceAttr("snowflake_view.test", "database", acc.TestDatabaseName),
-					resource.TestCheckResourceAttr("snowflake_view.test", "schema", acc.TestSchemaName),
-					resource.TestCheckResourceAttr("snowflake_view.test", "comment", "different comment"),
-					resource.TestCheckResourceAttr("snowflake_view.test", "copy_grants", "false"),
-					checkBool("snowflake_view.test", "is_secure", false),
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_View/basic_update"),
+				ConfigVariables: basicUpdate(rowAccessPolicy2.ID(), aggregationPolicy2, statement),
+				Check: assert.AssertThat(t, resourceassert.ViewResource(t, "snowflake_view.test").
+					HasNameString(id.Name()).
+					HasStatementString(statement).
+					HasDatabaseString(id.DatabaseName()).
+					HasSchemaString(id.SchemaName()).
+					HasCommentString("Terraform test resource"),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "aggregation_policy.#", "1")),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "aggregation_policy.0.policy_name", aggregationPolicy2.FullyQualifiedName())),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "aggregation_policy.0.entity_key.#", "1")),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "aggregation_policy.0.entity_key.0", "ROLE_NAME")),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "row_access_policy.#", "1")),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "row_access_policy.0.policy_name", rowAccessPolicy2.ID().FullyQualifiedName())),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "row_access_policy.0.on.#", "1")),
+					assert.Check(resource.TestCheckResourceAttr("snowflake_view.test", "row_access_policy.0.on.0", "ROLE_NAME")),
 				),
 			},
-			// change statement
+			// change statement and policies
```

Comment:
can we also check removing/adding policies externally?

Reply:
Good catch, done.